internal/services/ch/ch.go

```diff
@@ -0,0 +1,101 @@
+package ch
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/DIMO-Network/rewards-api/internal/config"
+)
+
+const (
+	defaultMaxExecutionTime                    = 5
+	defaultTimeoutBeforeCheckingExecutionSpeed = 2
+	// TimeoutErrCode is the error code returned by ClickHouse when a query is interrupted due to exceeding the max_execution_time.
+	TimeoutErrCode = int32(159)
+)
+
+// Client is a ClickHouse client that interacts with the ClickHouse database.
+type Client struct {
```

Comment:
Can we add a test for this guy? I know this entire repo is under-tested but no time like the present to improve.

I think Kevin has some infra [here](https://github.com/DIMO-Network/clickhouse-infra) but I've never really looked at it. I think Telemetry should provide ample examples of its use.